flink-connector-mongodb/src/main/java/org/apache/flink/connector/mongodb/sink/config/MongoWriteOptions.java

```diff
@@ -0,0 +1,205 @@
+                && maxRetryTimes == that.maxRetryTimes
+                && deliveryGuarantee == that.deliveryGuarantee
+                && Objects.equals(parallelism, that.parallelism);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(
+                bulkFlushMaxActions,
+                bulkFlushIntervalMs,
+                maxRetryTimes,
+                deliveryGuarantee,
+                parallelism);
+    }
+
+    public static MongoWriteOptionsBuilder builder() {
+        return new MongoWriteOptionsBuilder();
+    }
+
+    /** Builder for {@link MongoWriteOptions}. */
+    public static class MongoWriteOptionsBuilder {
+        private int bulkFlushMaxActions = BULK_FLUSH_MAX_ACTIONS.defaultValue();
+        private long bulkFlushIntervalMs = BULK_FLUSH_INTERVAL.defaultValue().toMillis();
+        private int maxRetryTimes = SINK_MAX_RETRIES.defaultValue();
+        private DeliveryGuarantee deliveryGuarantee = DeliveryGuarantee.AT_LEAST_ONCE;
+        private @Nullable Integer parallelism;
+
+        /**
+         * Sets the maximum number of actions to buffer for each bulk request. You can pass -1 to
+         * disable it. The default flush size 1000.
+         *
+         * @param numMaxActions the maximum number of actions to buffer per bulk request.
+         * @return this builder
+         */
+        public MongoWriteOptionsBuilder setBulkFlushMaxActions(int numMaxActions) {
+            checkState(
+                    numMaxActions == -1 || numMaxActions > 0,
+                    "Max number of buffered actions must be larger than 0.");
+            this.bulkFlushMaxActions = numMaxActions;
+            return this;
+        }
+
+        /**
+         * Sets the bulk flush interval, in milliseconds. You can pass -1 to disable it.
+         *
+         * @param intervalMillis the bulk flush interval, in milliseconds.
+         * @return this builder
+         */
+        public MongoWriteOptionsBuilder setBulkFlushIntervalMs(long intervalMillis) {
+            checkState(
+                    intervalMillis == -1 || intervalMillis >= 0,
+                    "Interval (in milliseconds) between each flush must be larger than "
+                            + "or equal to 0.");
+            this.bulkFlushIntervalMs = intervalMillis;
+            return this;
+        }
+
+        /**
+         * Sets the max retry times if writing records failed.
+         *
+         * @param maxRetryTimes the max retry times.
+         * @return this builder
+         */
+        public MongoWriteOptionsBuilder setMaxRetryTimes(int maxRetryTimes) {
+            checkArgument(
+                    maxRetryTimes >= 0, "The max retry times must be larger than or equal to 0.");
+            this.maxRetryTimes = maxRetryTimes;
+            return this;
+        }
+
+        /**
+         * Sets the wanted {@link DeliveryGuarantee}. The default delivery guarantee is {@link
+         * DeliveryGuarantee#NONE}
```

Comment:
The default shouldn't drop data.

Reply:
Sorry that's a typo in javadocs. The default guarantee is at-least-once.